hibernate-core/src/main/java/org/hibernate/id/enhanced/HiLoOptimizer.java

```diff
@@ -53,12 +53,78 @@
 public class HiLoOptimizer extends AbstractOptimizer {
 	private static final Logger log = Logger.getLogger( HiLoOptimizer.class );
 
-	private static class GenerationState {
+	protected static class GenerationState {
+		private final int incrementSize;
 		private IntegralDataTypeHolder lastSourceValue;
 		private IntegralDataTypeHolder upperLimit;
 		private IntegralDataTypeHolder value;
-	}
 
```

Comment:
The following changes involve callers referencing the new methods added to GenerationState, rather than accessing the fields directly.

An alternative is make the fields in GenerationState protected, instead of adding the methods. I really don't like that alternative though.